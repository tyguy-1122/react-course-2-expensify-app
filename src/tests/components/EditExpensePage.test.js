import React from 'react';
import { EditExpensePage } from '../../components/EditExpensePage';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';


test('should correctly render EditExpensePage', () => {
  const editExpense = jest.fn();
  const removeExpense = jest.fn();
  const history = { push: jest.fn() }
  const wrapper = shallow(
    <EditExpensePage
      expense={expenses[0]}
      removeExpense={removeExpense}
      editExpense={editExpense}
      history={history}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should handle editExpense', () => {
  const editExpense = jest.fn();
  const removeExpense = jest.fn();
  const history = { push: jest.fn() }
  const wrapper = shallow(
    <EditExpensePage
      expense={expenses[0]}
      removeExpense={removeExpense}
      editExpense={editExpense}
      history={history}
    />
  );
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(editExpense).toHaveBeenLastCalledWith(expenses[0].id, expenses[0]);
});

test('should handle removeExpense', () => {
  const editExpense = jest.fn();
  const removeExpense = jest.fn();
  const history = { push: jest.fn() }
  const wrapper = shallow(
    <EditExpensePage
      expense={expenses[0]}
      removeExpense={removeExpense}
      editExpense={editExpense}
      history={history}
    />
  );
  wrapper.find('button').simulate('click');
  expect(history.push).toHaveBeenLastCalledWith('/');
expect(removeExpense).toHaveBeenLastCalledWith({ id: expenses[0].id });
  expect(wrapper).toMatchSnapshot();
});