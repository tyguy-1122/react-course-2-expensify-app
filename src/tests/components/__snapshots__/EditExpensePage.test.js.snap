// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render EditExpensePage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;

exports[`should handle removeExpense 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
